azext_iot/common/shared.py

```diff
@@ -304,5 +309,33 @@ class SHAHashVersions(Enum):
     """
     Supported SHA types for generating the certificate thumbprint.
     """
+
     SHA1 = 1
     SHA256 = 256
+
+
+# Utility classes for nested edge config file values and device arguments
```

Comment:
I think these utility namedtuples should be located within the `iothub` namespace. This module is legacy and is in large part supporting the classic operations/* implementation style. Placing them either in `edge_device_config` or a new `.../iothub/common.py` would be more optimal.